Sources/Environment/Environment.swift

```diff
@@ -44,17 +44,18 @@ public struct Environment {
     public func all() -> [String: String] {
         var env = environ
         var pairs: [String: String] = [:]
-        while true {
-            guard let cval = env?.pointee else { break }
-            let pairString = String(cString: cval)
+        while let cpair = env?.pointee {
+            defer { env = env?.successor() }
+
+            let pairString = String(cString: cpair)
             let pair = pairString
-                .characters
+                .unicodeScalars
```

Comment:
Thanks so much for the analysis, I never looked this deep into it. At SwiftKey I've handled a lot of Unicode logic (also in Jay), so I know the difference in the terminology, but I wasn't sure about the exact meaning of the unicodeScalars property in String.

After thinking about this more, I'd like to keep it the way it is, with Characters. While I appreciate the performance difference, I value correctness more. E.g. consider the string `super=́key=value`. With splitting by unicodeScalars, you'd get `super` as the key and `key=value` with some accent prepended to it, but that's incorrect. The correct result will be returned if we use characters: `super=́key` as the key and `value` as value.

Thanks for helping me to make this decision deliberately by questioning it, it's a great point and I'm open to reconsideration if you find a usecase in which case it's a real bottleneck (I hope that doesn't happen though, as pulling all of the environment should not happen too often in an app).